std/experimental/allocator/package.d

```diff
@@ -1760,6 +1761,7 @@ if (is(T == class) || is(T == interface))
     auto support = (cast(void*) ob)[0 .. typeid(ob).initializer.length];
     destroy(p);
     alloc.deallocate(support);
+    p = null;
 }
 
 /// Ditto
```

Comment:
Can you also update the overload that takes a slice? Otherwise, LGTM.

Reply:
I'm on the side of those who think that testing `.length` should always be used on array / slice and not `null` comparison. But if anyone else confirm i'll add it.

Reply:
If you assign `null` to the slice, `.length` would return `0`, so code testing `.length` would still work.

Reply:
slice cant be disposed (even w/o the auto/ref + nullification). Just try this

```d
uint[] ua = Mallocator.instance.makeArray!uint([0,1]);
Mallocator.instance.dispose(ua[1..$]);
```

You'll get a nice memory error. I prefer not to touch to the overload made for arrays.

Reply:
That's by design. Disposing anything but the whole slice returned from `allocate` is undefined behavior.